File: medium/2_add_two_numbers.cpp
#include <iostream>


using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

// class Solution {
// public:
//     ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
//         return ;
//     }
// };

int main(){
    ListNode l1 = ListNode(3);
    for(auto i; i<=3; i++){
        *i
    }
    cout << l1.val<<endl;
    return 0;
}

File: medium/linked_list/2_add_two_numbers.py
import sys
sys.path.append('C:\\Users\\rockyo\\Desktop\\leetcode')
from ds import ListNode, createList, printList


class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        carry = 0
        o = ListNode()
        s = o
        while True:
            if l1==None and l2==None:
                if not carry ==0: 
                    val=0
                else:break
            elif l1==None: val=l2.val
            elif l2==None: val=l1.val
            else:          val = l1.val + l2.val

            sum   = (val+carry)
            val   = sum%10
            carry = sum// 10

            node = ListNode(val)
            if l1: l1=l1.next
            if l2: l2=l2.next

            o.next = node
            o = node
        return s.next


arr1 = [2,4,3]
arr2 = [5,6,4]
arr1 = [9,9,9,9,9,9,9]
arr2 = [9,9,9,9]

l = Solution().addTwoNumbers(createList(arr1), createList(arr2))
printList(l)

